pkg/utils/gardener/shootstate/shootstate.go

```diff
@@ -141,11 +141,15 @@ func computeGardenerData(
 		return nil, fmt.Errorf("failed compressing machine state data: %w", err)
 	}
 
-	return append(secretsToPersist, gardencorev1beta1.GardenerResourceData{
-		Name: v1beta1constants.DataTypeMachineState,
-		Type: v1beta1constants.DataTypeMachineState,
-		Data: runtime.RawExtension{Raw: machineStateJSONCompressed},
-	}), nil
+	if machineStateJSONCompressed != nil {
+		return append(secretsToPersist, gardencorev1beta1.GardenerResourceData{
+			Name: v1beta1constants.DataTypeMachineState,
+			Type: v1beta1constants.DataTypeMachineState,
+			Data: runtime.RawExtension{Raw: machineStateJSONCompressed},
+		}), nil
+	}
```

Comment:
```suggestion
	if machineStateJSONCompressed != nil {
		secretsToPersist = append(secretsToPersist, gardencorev1beta1.GardenerResourceData{
			Name: v1beta1constants.DataTypeMachineState,
			Type: v1beta1constants.DataTypeMachineState,
			Data: runtime.RawExtension{Raw: machineStateJSONCompressed},
		})
	}
```

Reply:
Done, tbh I was considering whether to do it like that before opening the PR :)